### PR TITLE
[BarClerk-948] - [FE] Integrate Get all clients/matter info in profil…

### PR DESCRIPTION
**Pull request by jasonclchua**
## Issue Link
- https://app.asana.com/0/1203234368773394/1203256371568948

## Definition of Done
- [ ] integrate client data in Client Profile page. Includes the ff:
  - [ ] client profile information - contribution, bail status, court, location, list of charges
  - [ ] client court appearances
  - [ ] client extensions
  - [ ] client funds information

## Pre-condition
- [ ] create sample client
- [ ] create GOL data and courtAppearances data for client

## Expected Output
  - [ ] when the user opens a client record from Dashboard Page, they will be redirected to the client's profile.
  - [ ] if client has extensions available, when user clicks on the client_extension dropdown, it will show the list of available extensions. Otherwise, if the client has no client_extension, 'no available data' message is shown on the dropdown. The client's funds are shown based on active client_extension from the dropdown. 
  - [ ] the client's court_appearances data are shown on the right portion of the page. If there are no court_appearances available for the client, 'no data available' message is shown instead.

## Screenshots/Recordings
[Main Dashboard _ Barclerk.webm](https://user-images.githubusercontent.com/118783823/206465681-0b77d107-6e4f-4b9c-935d-6008e769fde4.webm)

### CODE REVIEW

File: client/src/redux/client-profile/clientProfileSlice.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      })
+      .addCase(fetchSingleClientExtension.fulfilled, (state, action) => {
+        state.singleClientExtension = action.payload
+        state.isLoading = false
+      })
+  }
+})
+
+export const { reset } = clientProfile.actions
+export default clientProfile.reducer
```
Comment (impaulintech) [inline client/src/redux/client-profile/clientProfileSlice.ts]: Add new line

**Review by impaulintech — CHANGES_REQUESTED** [review]

**Review by jsvelte — COMMENTED** [review]
Please add custom skeleton loading style on initial load for reference please view this link: https://flowbite.com/docs/components/skeleton/ . Thank you 


And also please install this https://chrome.google.com/webstore/detail/screencastify-screen-vide/mmeijimgabbpbgpdklnllpncmdofkcpn?hl=en extension so that every time you make a PR you always send the Video Record and not the picture since this is an integration task

File: client/src/redux/client-profile/clientProfileSlice.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit'
+import type { PayloadAction } from '@reduxjs/toolkit'
+
+import { AxiosResponseError } from '~/shared/types'
+import { axios } from '~/shared/lib/axios';
+
+type InitialState = {
+  clientProfile: any
+  singleClientExtension: any
+  allClientExtensions: any
```
Comment (abduljalilpalala) [inline client/src/redux/client-profile/clientProfileSlice.ts]: Add it's proper types instead of `any`

**Review by abduljalilpalala — CHANGES_REQUESTED** [review]
I left some comments. Please check.

**Review by impaulintech — APPROVED** [review]
Resolve sir AJ's comment other than that, I approve.

**Review by jsvelte — APPROVED** [review]
# Approved 💯

**Review by abduljalilpalala — APPROVED** [review]

**Pull request merged**